File: app/auth/dtos/role.py
from pydantic import BaseModel, Field

from app.auth.dtos.permissions import PermissionDTO


class BaseRole(BaseModel):
    name: str
    description: str
    security_level: int
    permissions: list[PermissionDTO] = Field(default_factory=list)


class RoleDTO(BaseRole):
    id: int

File: frontend/src/types/api/role.ts
export interface PermissionDTO {
  id: number;
  name: string;
}

export interface RoleDTO {
  id: number;
  name: string;
  description: string;
  security_level: number;
  permissions: PermissionDTO[];
}

export interface ProjectRoleDTO {
  id: number;
  name: string;
  permissions: Record<string, unknown>;
}
